CMakeLists.txt

```diff
@@ -208,14 +208,33 @@ function(determine_version OUTPUT_VARIABLE)
 
   if(VERSION_MATCH)
     set(${OUTPUT_VARIABLE} ${CMAKE_MATCH_1} PARENT_SCOPE)
+    set(VERSION_STRING ${CMAKE_MATCH_1})
+    
   else()
     message(FATAL_ERROR "Invalid tag detected: ${NEW_VERSION}")
   endif()
+
+  string(REPLACE "-" ";" MULTIPASS_VERSION_COMPONENTS ${VERSION_STRING})
+  list(GET MULTIPASS_VERSION_COMPONENTS 0 MULTIPASS_SEMANTIC_VERSION)
+  set(MULTIPASS_SEMANTIC_VERSION ${MULTIPASS_SEMANTIC_VERSION})
+  set(MULTIPASS_BUILD_VERSION 0)
+  list(LENGTH MULTIPASS_VERSION_COMPONENTS MULTIPASS_VERSION_COMPONENTS_LENGTH)
+  if (MULTIPASS_VERSION_COMPONENTS_LENGTH GREATER 1)
+    list(GET MULTIPASS_VERSION_COMPONENTS 1 MULTIPASS_BUILD_VERSION_STR)
+    string(REGEX REPLACE "[^0-9]" "" MULTIPASS_BUILD_VERSION_NUMBER "${MULTIPASS_BUILD_VERSION_STR}")
```

Comment:
The CI runs sets the build version to `Setting build version to: 76638061597231`, cutting out the 1.16.0. IIUC release builds will then have a build version of `0`.

Reply:
So now, the build number will the the decimal representation of the commit hash, or 0 otherwise.